addon/index.js

```diff
@@ -3,7 +3,7 @@ import { assert } from '@ember/debug';
 export function taskFor(task) {
   assert(
     `${task} does not appear to be a task!`,
-    task && typeof task.perform === 'function'
+    task && (typeof task === 'function' || typeof task.perform === 'function')
```

Comment:
This doesn't work for encapsulated tasks. Unfortunately, if we want this to work I think we probably just have to give up on having an assertion at all. The original intent for this is to guard the fact that, despite allowed by the types, passing a normal generator or async function here does not magically turn it into a task, and this really only works if you have the decorator pre-applied. Now if we are purposefully allowing that, I think we'll just have to remove the assertion. We _could_ update it at check specifically for the kind of functions or POJOs that are allowed by the decorator, but at that point we would just be duplicating what TypeScript already checked for.

Reply:
It actually works just fine because an encapsulated task descriptor is an object with a function named `perform`. I've added a test proving so. 😄